logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageMapper.kt

```diff
@@ -67,93 +67,121 @@ class MessageMapperImpl(
-                isSelfMessage = message.isSelfMessage,
-                expectsReadConfirmation = message.expectsReadConfirmation
-            )
-
-            is Message.System -> MessageEntity.System(
-                id = message.id,
-                content = message.content.toMessageEntityContent(),
-                conversationId = message.conversationId.toDao(),
-                date = message.date.toInstant(),
-                senderUserId = message.senderUserId.toDao(),
-                status = status,
-                visibility = visibility,
-                senderName = message.senderUserName,
-            )
+            is Message.Regular -> mapFromRegularMessage(message, status, visibility)
+            is Message.System -> mapFromSystemMessage(message, status, visibility)
         }
     }
 
+    private fun mapFromRegularMessage(
+        message: Message.Regular,
+        status: MessageEntity.Status,
+        visibility: MessageEntity.Visibility
+    ) =
+        MessageEntity.Regular(
+            id = message.id,
+            content = toMessageEntityContent(message.content),
+            conversationId = message.conversationId.toDao(),
+            date = message.date.toInstant(),
+            senderUserId = message.senderUserId.toDao(),
+            senderClientId = message.senderClientId.value,
+            status = status,
+            editStatus = when (message.editStatus) {
+                is Message.EditStatus.NotEdited -> MessageEntity.EditStatus.NotEdited
+                is Message.EditStatus.Edited -> MessageEntity.EditStatus.Edited(message.editStatus.lastTimeStamp.toInstant())
+            },
+            expireAfterMs = message.expirationData?.let { it.expireAfter.inWholeMilliseconds },
+            selfDeletionStartDate = message.expirationData?.let {
+                when (val status = it.selfDeletionStatus) {
+                    is Message.ExpirationData.SelfDeletionStatus.Started -> status.selfDeletionStartDate
+                    is Message.ExpirationData.SelfDeletionStatus.NotStarted -> null
+                }
+            },
+            visibility = visibility,
+            senderName = message.senderUserName,
+            isSelfMessage = message.isSelfMessage,
+            expectsReadConfirmation = message.expectsReadConfirmation
+        )
+
+    private fun mapFromSystemMessage(
+        message: Message.System,
+        status: MessageEntity.Status,
+        visibility: MessageEntity.Visibility
+    ) = MessageEntity.System(
+        id = message.id,
+        content = message.content.toMessageEntityContent(),
+        conversationId = message.conversationId.toDao(),
+        date = message.date.toInstant(),
+        senderUserId = message.senderUserId.toDao(),
+        status = status,
+        visibility = visibility,
+        senderName = message.senderUserName,
+        expireAfterMs = message.expirationData?.let { it.expireAfter.inWholeMilliseconds },
```

Comment:
So the reason why we care about expiration for system messages is because pings are system messages? 

Not for this PR but I think it makes sense to stop mapping pings to system messages because really they aren't system messages. They are just another kind of message a user can send but we choose to treat them as system message because of how they are displayed?

Reply:
100% agree